docs/designs/xref.md

```diff
@@ -4,6 +4,21 @@ Besides using file path to link to another file, DocFX also allows you to give a
 - API reference documentation is usually auto generated so it's difficult to find its file path.
 - You want to reference to files in another project without need to know its file structure.
 
+## Feature Requirements/Scenarios
+- Define an internal UID and reference to this UID within the current repository
+- Define an internal UID with versioning and reference to this UID without versioning within the current repository
+- Define multiple UID with same value internally with different versioning and reference to this UID without versioning within the current repository
+    - If all versionings for this UID are within the same product, take the one with highest versioning respecting the referencing file
+    - If all versionings for this UID are from different products, take the highest one base on proudct namme alphabetically
+- Reference to an external UID without versioning
+    - The href of UID is from the same host name as the referencing repository
+    - The href of UID is from a different host name as the referencing repository
+        - If the current branch is live branch, the outout path could be 'review.docs', but the resolved url is 'docs', it will go to another site clicking on the url
+        - If the current branch is from a different site other than 'docs', we need to prioritize the uid from this site firstly if same uid defined in this site and 'docs' 
+- Reference to an external UID with versioning
```

Comment:
does this item mean the referenced Uid has just one version?

Reply:
The next item(UID with multiple versionings) is not in scope for versioning during migration from v2 to v3